Begin to use QStrings

sithlord48 opened the pull request:
I had intended for this to enable QStrings All over. 
But there are so many windows changes from Qt defining UNICODE  that this pr mostly enables that

 - Use QStrings in Logs and Daemon
 - Remove ArchString , use QString for unicode to / from utf8 / latin1
 - Correct build issues on windows when linking Qt due to strings being unicode now

TODO: 
- [x] Find why Window Daemon spits garbage messages
- [x] Fix `could not send close event to process` bug (slow Core restart)
- [x] Test CI package (following Windows memory bugs)
- [x] Squash commits (added by @nbolton)

nbolton reviewed (changes requested):
It appears this PR causes the Windows daemon to be stuck in 'Starting' state.

sithlord48 commented:
> It appears this PR causes the Windows daemon to be stuck in 'Starting' state.

I fixed this just now , i changed our `ArchMiscWindows::findLoadedModule` to take a `wchar_t *` that  was causing the daemon to not find the msvc dlls. I've reverted that part to use a good ol'  `char *` again

nbolton commented:
> Did you try the CI artifacts ? it worked and was able to connect to my server from my window vm.

CI artefact did not work for me. GUI doesn't connect to the daemon, and seems unresponsive. After a while some garbage chinese chars are printed to the log.

![image](https://github.com/user-attachments/assets/db9d759f-2266-427a-827e-c185e6e4cae3)

My `C:\Program Files\Deskflow\deskflow-daemon.log` file contains:
```
㉛㈰ⴵ㘰〭吵ㄲㄺ㨰㜴⁝义但›楢浴灡›㜱㈴㡸㔷㌠ല嬊〲㔲〭ⴶ㔰㉔㨱〱㐺崷䤠䙎㩏挠湯敶瑲椠慭敧映潲㩭搠灥桴㌽′潣灭㌽਍㉛㈰ⴵ㘰〭吵ㄲㄺ㨰㜴⁝低䕔›歳灩楰杮挠楬扰慯摲琠慲獮敦⁲敢慣獵⁥桴⁥汣灩潢牡⁤潣瑮湥獴攠捸敥獤琠敨㌠䴠⁂楳敺氠浩瑩猠瑥戠⁹桴⁥敳癲牥਍㉛㈰ⴵ㘰〭吵ㄲㄺ㨰㜴⁝义但›楢浴灡›㜱㈴㡸㔷㌠ല嬊〲㔲〭ⴶ㔰㉔㨱〱㐺崷䤠䙎㩏挠湯敶瑲椠慭敧映潲㩭搠灥桴㌽′潣灭㌽਍㉛㈰ⴵ㘰〭吵ㄲㄺ㨰㜴⁝低䕔›歳灩楰杮挠楬扰慯摲琠慲獮敦⁲敢慣獵⁥桴⁥汣灩潢牡⁤潣瑮湥獴攠捸敥獤琠敨㌠䴠⁂楳敺氠浩瑩猠瑥戠⁹桴⁥敳癲牥਍〺崵丠呏㩅
```

sithlord48 commented:
> [2025-09-08T12:19:03] ERROR: standard exception on thread 0x0000bad0: resource deadlock would occur: resource deadlock would occur


I seen this in another issue this morning https://github.com/deskflow/deskflow/issues/7329#issuecomment-3265141505

nbolton commented on src/lib/platform/XWindowsUtil.cpp:
```diff
@@ -5,12 +5,12 @@
  */
 
-#include "platform/XWindowsUtil.h"
+#include "base/Log.h" //Include First
```
Maybe add the word `// HACK` to the start of the comment.

nbolton commented on src/lib/platform/MSWindowsDesks.cpp:
I imagine this is just the tip of the iceberg. Anywhere we use `%s` specifier with `wstring` we will get only the 1st char printed.

nbolton commented on cspell.json:
```diff
@@ -46,6 +46,7 @@
     "logonui",
     "Lysytsia",
     "macdeployqt",
+    "mojibake",
```
I learned a new word! :D 
It's a Japanese word pronounced "mo-jeh-ba-keh"

nbolton commented:
> [ ] Fix `could not send close event to process` bug (slow Core restart)

Looking into this now.

sithlord48 commented:
> Would you like me to squash them into your original commit? 

 Yes please do.

nbolton commented:
<img width="531" height="336" alt="image" src="https://github.com/user-attachments/assets/e99a5065-c72f-44eb-ba04-094b6407d928" />
